src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+
 public class WorkWithFile {
+    private int supply = 0;
+    private int buy = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        readFile(fromFileName);
+        writeFile(toFileName);
+        cleanBuffered();
+    }
+
+    private void writeFile(String toFileName) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
+            bufferedWriter.write("supply," + supply
+                    + "\r\n" + "buy," + buy
+                    + "\r\n" + "result," + (supply - buy));
+        } catch (Exception e) {
+            throw new RuntimeException("Can't read data from the file " + toFileName, e);
+        }
+    }
+
+    private void readFile(String fromFileName) {
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
```

Comment:
Avoid nested try blocks. The BufferedReader should be initialized in its own try-with-resources block.

Reply:
Fix comment